trie/smt.py

```diff
@@ -0,0 +1,177 @@
+from eth_utils import (
+        keccak,
+        to_int,
+    )
+
+from trie.constants import (
+    BLANK_NODE,
+)
+
+from trie.exceptions import (
+        ValidationError,
+    )
+
+from trie.validation import (
+    validate_is_bytes,
+    validate_length,
+)
+
+
+class SparseMerkleTree:
+
+    def __init__(self, keysize=32):
+        # Ensure we can support the given height
+        if 1 > keysize or keysize > 32:
+            raise ValidationError("Keysize must be number of bytes in range [0, 32]")
+
+        self.keysize = keysize  # keysize in bytes
+        self.height = keysize * 8  # height is number of bits in key
+        
+        # Initialize an empty tree with one branch
+        self.db = {}
```

Comment:
I'd suggest following the same pattern as `HexaryTrie` and having the `db` be something that is passed in (so we can use non-memory based databases).

Reply:
I built a class method for that purpose, since you have to initialize the database to a particular depth, with the root hash at the top. I could have that in the constructor, but I wanted to keep it separate to make it clearer that there is the potential for inconsistency if it accepts an outside db.